include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -55,40 +60,92 @@ template<typename T> // integers operate on 2s complement so same op for signed
 [[vk::ext_instruction(spv::OpAtomicIAdd)]]
 T atomicIAdd([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
 
+template<typename T, typename Ptr_T> // DXC Workaround
+[[vk::ext_instruction(spv::OpAtomicIAdd)]]
+typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, T>::type atomicIAdd(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
```

Comment:
I'd slap extra checks (via enable_if) that `T` is a 32bit integer type on the Add, SMin, UMin and Max

Reply:
actually everything except atomic Load, Store, Exchange dictate that the type must be `integral` and not a boolean

In our case lets check for `T` being same as `uint32_t` and `int32_t`, since the 16bit and 64bit overloads will probably need some special capabilitiy attributes and hence be separate overloads

Reply:
also one more note, you're already in `nbl::hlsl::` no need for that prefix

and add an ` enable_if_t` alias template to type_traits.hlsl while you're at it so there's less noise in the function defs

Reply:
Done